lib/puppet/indirector/node/write_only_yaml.rb

```diff
@@ -11,7 +11,12 @@
 # @api private
 #
 class Puppet::Node::WriteOnlyYaml < Puppet::Indirector::Yaml
-  desc "Store node information as flat files, serialized using YAML,
+  def initialize
+    Puppet.deprecation_warning('Puppet::Node::WriteOnlyYaml is deprecated and will be removed in a future release of Puppet.')
+    super
+  end
```

Comment:
If we create a new WriteOnlyYaml terminus for each request, then this will spam the logs. Can we warn once (but also see comment below)?